### ISSUE TITLE
Solution with application and static library fails to link

### ISSUE THREAD

**Issue by suprstefan** [Microsoft/VSLinux#63]
I've created a simple solution ([test03.zip](https://github.com/Microsoft/VSLinux/files/520427/test03.zip)) with two projects: an application and a static library that the application uses. It compiles fine but fails to link. I've added the static library as a reference in the application. 

When building I get the follow output:

```
1>------ Build started: Project: lib, Configuration: Debug ARM ------
1>  Validating architecture
1>  Validating sources
1>  Copying sources remotely
1>  Starting remote build
1>  Compiling sources:
1>  foo.cpp
1>  Archiving objects
1>  lib.vcxproj -> C:\dev\rpi\test03\lib\bin\ARM\Debug\liblib.a
2>------ Build started: Project: app, Configuration: Debug ARM ------
2>  Validating architecture
2>  Validating sources
2>  Copying sources remotely
2>  Starting remote build
2>  Compiling sources:
2>  main.cpp
2>  Linking objects
2>C:\dev\rpi\test03\app\obj\ARM\Debug\main.o : error :
2>C:\dev\rpi\test03\app\main.cpp(7): error : undefined reference to `add(int, int)'
2>collect2 : error : ld returned 1 exit status
========== Build: 1 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

Doing a `ps aux` when linking it says that it runs the following:

```
g++ -o /home/pi/projects/app/bin/ARM/Debug/app.out -Wl,--no-undefined -Wl,-z,relro -Wl,-z,now -Wl,-z,noexecstack /home/pi/projects/app/obj/ARM/Debug/main.o
```

This is slightly odd since the lib isn't part of the command line, but looking at "Command line" in visual studio it says:

```
-o"C:\dev\rpi\test03\app\bin\ARM\Debug\app.out" -Wl,-z,relro "C:\dev\rpi\test03\lib\bin\ARM\Debug\liblib.a" -Wl,-z,noexecstack -Wl,--no-undefined -Wl,-z,now 
```

It seems like "liblib.a" gets lost in translation. If I add it to the command line and run it on the pi it links and works great.

Versions:
- VS2015 Update 3
- Linux tools 1.0.5
- g++ (Raspbian 4.9.2-10) 4.9.2 running on a raspberry pi 1

**Comment by stanthomas** [comment]
Looks like you've specified a path to the library on your Windows system `"C:\dev\rpi\test03\lib\bin\ARM\Debug\liblib.a"`, which g++ would have trouble with if it hadn't been removed by VCLinux :wink:

You can specify the library to the application project either via Link/General/Additional Library Directories **and** Link/Input/Library Dependencies, in which case give the library name without `lib` prefix and `.a` extension. Or by giving full path to the library in Link/Input/Additional Dependencies (which seems to be the route you've chosen). Paths should be Linux file paths on the remote target.

**Comment by suprstefan** [comment]
I didn't add it, VS did when I added a reference to "lib" from "app". Is that not supposed to work? That seems like a quite basic case to support. The -o is also pointing to my c: which made me think that the extension does some kind of translation.

**Comment by stanthomas** [comment]
I might be wrong but I don't think many C++ projects use references to link libraries -  bit of a C# thing. But it's there so I guess it ought to work.

For now though you can build your application by adding the library thru the project settings.

**Comment by suprstefan** [comment]
I'd say that you're wrong: https://msdn.microsoft.com/en-us/library/47w1hdab.aspx

It used to be a C# only thing up until VS2013-ish. Now it seems to be the recommended way for VS2015 solutions as well.

I can work around it by adding the path and lib-file manually, but it still seems like a bug to me.

**Comment by Unarmed1000** [comment]
Yes clearly a bug that ought to be solved ASAP.

**Comment by akaltar** [comment]
The option was available in VS2010 in the project property pages.

I think that other similar projects(android) have this feature working too.

**Comment by levhaikin** [comment]
I've stumbled into the same problem. Have several static projects, and one so. The so fails to link, and it doesn't have the static libraries on the link path.
Are there any (fix-in-progress hopefully) news regarding this issue?

**Comment by itodirel** [comment]
@yuehuang010 Felix assigning to you as you fixed the P2Ps recently, this should been fixed too by that fix

**Comment by lukka** [comment]
This issue is fixed in the next VS preview (after 15.6 is released). Please re-open if you experience any issue related to project to project reference.
